tests/devapps/MSI/HelperService/MSIHelperService/Helper/MSIHelper.cs

```diff
@@ -0,0 +1,370 @@
+        internal static string? s_functionAppEnvCode = Environment.GetEnvironmentVariable("functionAppEnvCode");
+        internal static string? s_functionAppMSICode = Environment.GetEnvironmentVariable("functionAppMSICode");
+        internal static string? s_webhookLocation = Environment.GetEnvironmentVariable("webhookLocation");
+        internal static string? s_oMSAdminClientID = Environment.GetEnvironmentVariable("OMSAdminClientID");
+        internal static string? s_oMSAdminClientSecret = Environment.GetEnvironmentVariable("OMSAdminClientSecret");
+
+        //Protected from this API so the getEnvironmentVariable does not return these
+        internal static string[] s_protectedEnvironmentVariables = new string[] {
+            "requestAppID",
+            "requestAppSecret",
+            "functionAppUri",
+            "functionAppEnvCode",
+            "functionAppMSICode",
+            "webhookLocation",
+            "OMSAdminClientID",
+            "OMSAdminClientSecret",
+            "path"
+        };
+
+        internal const string Authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47";
+
+        //Enum for HTTP Error Response Codes
+        internal enum HTTPErrorResponseCode : int
+        {
+            Status200OK = StatusCodes.Status200OK,
+            Status201Created = StatusCodes.Status201Created,
+            Status400BadRequest = StatusCodes.Status400BadRequest,
+            Status404NotFound = StatusCodes.Status404NotFound,
+            Status500InternalServerError = StatusCodes.Status500InternalServerError,
+            Status503ServiceUnavailable = StatusCodes.Status503ServiceUnavailable
+        }
+
+        //Azure Resources
+        internal enum AzureResource
+        {
+            webapp,
+            function,
+            vm,
+            azurearc,
+            servicefabric,
+            cloudshell
+        }
+
+        /// <summary>
+        /// Gets the Environment Variable from the Azure Web App
+        /// </summary>
+        /// <param name="variableName"></param>
+        /// <param name="logger"></param>
+        /// <returns>Returns the environment variable</returns>
+        public static string? GetWebAppEnvironmentVariable(
+            string variableName,
+            ILogger logger)
+        {
+            //On the Web app we have stored some environment variables, those variables 
+            //hold values that are used to call into several other azure resources
+            //Hide the App ID and App Secret and other variableds in this API
+            //these are used to get token for other azure test resources and stored in the
+            //web apps environment variables
+
+            logger.LogInformation("GetWebAppEnvironmentVariable Function called.");
+
+            return
+                s_protectedEnvironmentVariables.Any(s => s.Contains(variableName, StringComparison.OrdinalIgnoreCase)) ?
```

Comment:
Instead of sending NULL, can you send a constant like "REDACTED" or smth like that? That would make the test more useful I think, because having a bad env var value is more useful than not having an env var.

Reply:
it's protected